### PR TITLE
Data Generator and Load Tester

### PR DESCRIPTION
**Pull request by alvinalexander**
I am still working on a read me for the load tester

### CODE REVIEW

File: bin/debug/intake_single_user.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 
 if __name__ == '__main__':
     np.random.seed(61297777)
-
+
```
Comment (shankari) [inline bin/debug/intake_single_user.py]: Please revert

File: conf/net/api/webserver.conf.sample [hunk 2]
```diff
@@ -13,4 +13,5 @@
     "timeout" : "3600",
     "auth": "skip"
   }
+
```
Comment (shankari) [inline conf/net/api/webserver.conf.sample]: please revert

File: docker/Dockerfile [hunk 3]
```diff
@@ -6,7 +6,7 @@ MAINTAINER K. Shankari (shankari@eecs.berkeley.edu)
 WORKDIR /usr/src/app
 
 # clone from repo
-RUN git clone https://github.com/e-mission/e-mission-server.git .
+RUN git clone https://github.com/alvinalexander/e-mission-server.git .
```
Comment (shankari) [inline docker/Dockerfile]: please revert

File: docker/docker-compose.yml [hunk 4]
```diff
@@ -39,4 +39,4 @@ networks:
   emission:
 
 volumes:
-  mongo-data:
\ No newline at end of file
+  mongo-data:
```
Comment (shankari) [inline docker/docker-compose.yml]: why do you need this?

File: docker/start_script.sh [hunk 5]
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/env bash
 #Configure web server
-
```
Comment (shankari) [inline docker/start_script.sh]: revert unrelated changes

File: docker/start_script.sh [hunk 6]
```diff
@@ -27,3 +26,4 @@ source activate emission
 
 # launch the webapp
 ./e-mission-py.bash emission/net/api/cfc_webapp.py
+
```
Comment (shankari) [inline docker/start_script.sh]: revert unrelated changes

File: emission/analysis/plotting/geojson/test_geojson.py [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import emission.core.get_database as edb
```
Comment (shankari) [inline emission/analysis/plotting/geojson/test_geojson.py]: what does this script do? If it is a utility function, maybe it should be in `bin` instead

File: emission/net/ext_service/geocoder/nominatim.py [hunk 8]
```diff
@@ -14,14 +15,22 @@
 from pygeocoder import Geocoder as pyGeo  ## We fall back on this if we have to
 
 try:
-    googlemaps_key_file = open("conf/net/ext_service/googlemaps.json")
+    script_dir = os.path.dirname(__file__)
```
Comment (shankari) [inline emission/net/ext_service/geocoder/nominatim.py]: why do you need these changes?

File: emission/net/ext_service/otp/test_otp.py [hunk 9]
```diff
@@ -0,0 +1,144 @@
+import unittest
```
Comment (shankari) [inline emission/net/ext_service/otp/test_otp.py]: tests should be in `emission/tests` so that they can be included in the automated tests

File: emission/net/usercache/formatters/android/motion_activity.py [hunk 10]
```diff
@@ -22,6 +22,8 @@ def format(entry):
     fc.expand_metadata_times(metadata)
     formatted_entry.metadata = metadata
 
+    logging.info('*** Motion Data write_ts: %d' % metadata.write_ts)
```
Comment (shankari) [inline emission/net/usercache/formatters/android/motion_activity.py]: please comment out logging, we already have enough!

File: emission/simulation/test_prob140.py [hunk 11]
```diff
@@ -0,0 +1,30 @@
+import unittest
```
Comment (shankari) [inline emission/simulation/test_prob140.py]: again, unittests go into `emission/tests`

File: emission/simulation/test_user.py [hunk 12]
```diff
@@ -0,0 +1,90 @@
+import unittest
```
Comment (shankari) [inline emission/simulation/test_user.py]: ditto

File: emission/storage/pipeline_queries.py [hunk 13]
```diff
@@ -252,7 +252,7 @@ def get_time_range_for_stage(user_id, stage):
 def get_current_state(user_id, stage):
     curr_state_doc = edb.get_pipeline_state_db().find_one({"user_id": user_id,
                                                             "pipeline_stage": stage.value})
-    # logging.debug("returning curr_state_doc  %s for stage %s " % (curr_state_doc, stage))
+    logging.debug("returning curr_state_doc  %s for stage %s " % (curr_state_doc, stage))
```
Comment (shankari) [inline emission/storage/pipeline_queries.py]: restore comment

File: docker/docker-compose.yml (hunk 4 above)
Comment (alvinalexander) [reply docker/docker-compose.yml]: Not sure what you mean. why do I need the mongo-data volume?

File: emission/analysis/plotting/geojson/test_geojson.py (hunk 7 above)
Comment (alvinalexander) [reply emission/analysis/plotting/geojson/test_geojson.py]: I added this when I was debugging the data generator. it's just a copy paste from the Timeseries_Sample notebook.

File: bin/debug/intake_single_user.py [hunk 14]
```diff
@@ -30,4 +30,4 @@
     else:
         sel_uuid = ecwu.User.fromEmail(args.user_email).uuid
 
-    epi.run_intake_pipeline("single", [sel_uuid])
+    epi.run_intake_pipeline("single", [sel_uuid])
```
Comment (shankari) [inline bin/debug/intake_single_user.py]: what is the change here?

File: conf/net/api/webserver.conf.sample [hunk 15]
```diff
@@ -13,4 +13,4 @@
     "timeout" : "3600",
     "auth": "skip"
   }
-}
+}
```
Comment (shankari) [inline conf/net/api/webserver.conf.sample]: what is the change here?

File: docker/Dockerfile [hunk 16]
```diff
@@ -40,4 +40,4 @@ RUN chmod u+x /usr/src/app/start_script.sh
 
 EXPOSE 8080
 
-CMD ["/bin/bash", "/usr/src/app/start_script.sh"]
+CMD ["/bin/bash", "/usr/src/app/start_script.sh"]
```
Comment (shankari) [inline docker/Dockerfile]: what is the change here?

File: docker/start_script.sh [hunk 17]
```diff
@@ -26,4 +26,4 @@ cat conf/net/api/webserver.conf
 source activate emission
 
 # launch the webapp
-./e-mission-py.bash emission/net/api/cfc_webapp.py
+./e-mission-py.bash emission/net/api/cfc_webapp.py
```
Comment (shankari) [inline docker/start_script.sh]: what is the change here?

File: emission/analysis/plotting/geojson/test_geojson.py (hunk 7 above)
Comment (shankari) [reply emission/analysis/plotting/geojson/test_geojson.py]: ok, so you can either remove it from the commit, or move it into `bin` which is the location for standalone scripts as opposed to modules that are to be integrated into other scripts.

File: emission/core/get_database.py [hunk 18]
```diff
@@ -21,6 +21,7 @@
 
 print("Connecting to database URL "+url)
 _current_db = MongoClient(url).Stage_database
+#config_file.close()
```
Comment (shankari) [inline emission/core/get_database.py]: revert

File: emission/net/ext_service/geocoder/nominatim.py (hunk 8 above)
Comment (shankari) [reply emission/net/ext_service/geocoder/nominatim.py]: @alvinalexander why do you need these changes?

File: emission/net/ext_service/otp/otp.py [hunk 19]
```diff
@@ -98,104 +116,250 @@ def get_all_trips(self, _id, user_id, trip_id):
...
+    )
+    return end_loc
+
+def opt_mode_to_motiontype(opt_mode):
+    #TODO: this needs to be made more sophisticated. This should include all modes supported by OTP client
+    # and emission server.
+    mapping = {
+        'CAR': ecwm.MotionTypes.IN_VEHICLE,
+        'RAIL': ecwm.MotionTypes.IN_VEHICLE,
+        'WALK': ecwm.MotionTypes.WALKING
```
Comment (shankari) [inline emission/net/ext_service/otp/otp.py]: so are these the only current trip modes supported?

File: emission/net/ext_service/otp/planner.json [hunk 20]
```diff
@@ -1,3 +1,3 @@
 {
-	"open_trip_planner_instance_address" : "http://54.162.140.55:8080"
+	"open_trip_planner_instance_address" : "http://18.209.111.117"
```
Comment (shankari) [inline emission/net/ext_service/otp/planner.json]: did you add instructions about how to launch OTP using your docker instance + load data?

File: emission/net/usercache/formatters/android/motion_activity.py (hunk 10 above)
Comment (shankari) [reply emission/net/usercache/formatters/android/motion_activity.py]: @alvinalexander please comment out logging in the committed version

File: emission/simulation/test_prob140.py (hunk 11 above)
Comment (shankari) [reply emission/simulation/test_prob140.py]: @alvinalexander please move unittests to `emission/tests` and ensure that they pass with `runAllTests.sh`

File: bin/debug/intake_single_user.py (hunk 14 above)
Comment (alvinalexander) [reply bin/debug/intake_single_user.py]: How exactly do you I revert back to the old version of these files?

Comment (shankari) [reply bin/debug/intake_single_user.py]: if these changes are all in a few commits, you can use `git revert`. Otherwise, you can clone a new repo and copy over the original files

https://stackoverflow.com/a/6971775/4040267

**Comment by shankari** [comment]
Thanks to @njriasan for merging this in #644
